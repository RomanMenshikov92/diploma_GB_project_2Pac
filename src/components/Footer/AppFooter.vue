<template lang="pug">
footer.footer
  .container.footer__container
    .footer__logo
      Logo
    .footer__copyright
      Copyright(:copyright="copyrightItem")
</template>

<script lang="ts">
import { computed, defineComponent } from "vue";
import Logo from "@/components/Common/Logo.vue";
import Copyright from "@/components/Footer/Ð¡opyright.vue";
import { useCopyrightStore } from "@/stores/Common/useCopyrightStore";

export default defineComponent({
  name: "AppFooter",
  components: {
    Logo,
    Copyright,
  },
  setup() {
    const copyrightStore = useCopyrightStore();
    const copyrightItem = computed(() => copyrightStore.getCopyrightItem);

    return {
      copyrightItem,
    };
  },
});
</script>

<style lang="sass"></style>
