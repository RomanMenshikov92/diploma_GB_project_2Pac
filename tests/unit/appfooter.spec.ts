import { shallowMount, mount } from "@vue/test-utils";
import AppFooter from "@/components/Footer/AppFooter.vue";
import Logo from "@/components/Common/Logo.vue";
import Copyright from "@/components/Footer/Сopyright.vue";
import { Pinia, createPinia } from "pinia";

jest.mock("@/components/Common/Logo.vue", () => ({
  name: "Logo",
  render: jest.fn(),
}));

jest.mock("@/components/Footer/Сopyright.vue", () => ({
  name: "Copyright",
  render: jest.fn(),
}));

describe("AppFooter.vue", () => {
  it("renders in AppFooter components: Logo", () => {
    const pinia: Pinia = createPinia();

    const wrapper = shallowMount(AppFooter, {
      global: {
        plugins: [pinia],
      },
    });
    expect(wrapper.findComponent(Logo).exists()).toBe(true);
  });

  it("renders in AppFooter components: Copyright", () => {
    const pinia: Pinia = createPinia();

    const wrapper = shallowMount(AppFooter, {
      global: {
        plugins: [pinia],
      },
    });
    expect(wrapper.findComponent(Copyright).exists()).toBe(true);
  });

  it("renders correct Pug in footer layout", () => {
    const pinia: Pinia = createPinia();

    const wrapper = mount(AppFooter, {
      global: {
        plugins: [pinia],
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
