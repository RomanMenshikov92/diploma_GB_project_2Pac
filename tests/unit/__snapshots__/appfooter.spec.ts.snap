// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppFooter.vue renders correct Pug in footer layout 1`] = `
<footer class="footer">
  <div class="container footer__container">
    <div class="footer__logo">
      <!---->
    </div>
    <div class="footer__copyright">
      <!---->
    </div>
  </div>
</footer>
`;
